models/src/main/kotlin/motif/models/FactoryMethod.kt

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2018 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package motif.models
+
+import motif.DoNotCache
+import motif.Expose
+import motif.ast.IrClass
+import motif.ast.IrMethod
+import motif.ast.IrParameter
+import javax.inject.Inject
+
+/**
+ * [Wiki](https://github.com/uber/motif/wiki#factory-methods)
+ */
+sealed class FactoryMethod(
+        val method: IrMethod,
+        val objects: Objects) {
+
+    abstract val returnType: ReturnType
+    abstract val parameters: List<Parameter>
+
+    val isStatic = method.isStatic()
+    val isCached = !method.hasAnnotation(DoNotCache::class)
```

Comment:
What's the lookup time on these? Are they being used once, should we cache the values here?

Reply:
This currently does cache the value. Are you suggesting not caching?

Reply:
Whoops, was multitasking and misread it. All good from my side.

Reply:
Speaking of my experience, I noticed that in most cases the dependencies in mono repo are cached in its own scopes. I would vote up to cache the value.

Reply:
Hi @TonyTangAndroid - Objects are cache by default in Motif. This annotation is to opt out of that behavior.

Reply:
Got it. Understood.